### DISCUSSION TITLE
Property 'api' does not exist on type 'SupabaseAuthClient'.ts

### DISCUSSION THREAD

**Discussion by Zacklau922** [supabase/supabase#20331] [Questions]
Im using Nextjs in this project, I want to invite an email of a user once the event is confirmed, but encounter error:
--- Property 'api' does not exist on type 'SupabaseAuthClient'.ts(2339)

This is how I do it:

```
import { createClient } from '@supabase/supabase-js'

const supabase = createClient(SUPABASE_URL, process.env.SERVICE_KEY);

const { data: inviteData, error: inviteError } = await supabase.auth.api.inviteUserByEmail('user@example.com')

if (inviteError) {
  console.error('Error inviting user:', inviteError)
  alert('Failed to invite user.')
  errorOccurred = true
}
```

Before this, I was using this, but this is a sign-up, not through inviting users, I was unable to modify the template in this email. 
This is how I do it:

```
import { createClientComponentClient } from "@supabase/auth-helpers-nextjs";

  const supabase = createClientComponentClient();

      const { data: newUser, error: newUserError } = await supabase.auth.signUp(
        {
          email: payment.email,
          password: "123456",
        }
      );
```

How do I register an email into a service role? Any example or doc on this? 

Thanks

**Comment by GaryAustin1** [comment]
You are using v1 supabase-js syntax.    It is now admin instead of api.   https://supabase.com/docs/reference/javascript/admin-api
